stable/rocketchat/templates/deployment.yaml

```diff
@@ -3,25 +3,31 @@ kind: Deployment
 metadata:
   name: {{ template "rocketchat.fullname" . }}
   labels:
-    app: {{ template "rocketchat.name" . }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
+    app.kubernetes.io/name: {{ include "rocketchat.name" . }}
+    helm.sh/chart: {{ include "rocketchat.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
 spec:
   replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "rocketchat.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
   template:
     metadata:
       labels:
-        app: {{ template "rocketchat.name" . }}
-        release: "{{ .Release.Name }}"
+        app.kubernetes.io/name: {{ include "rocketchat.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      annotations:
+        checksum/secret: {{ include (print .Template.BasePath "/secret.yaml") . | sha256sum }}
     spec:
       {{- if .Values.securityContext.enabled }}
       securityContext:
         fsGroup: {{ .Values.securityContext.fsGroup }}
         runAsUser: {{ .Values.securityContext.runAsUser }}
       {{- end }}
       containers:
-      - name: {{ template "rocketchat.fullname" . }}
+      - name: rocketchat
```

Comment:
wouldn't this create duplicate containers if you install 2 rocketchat instances in the same namespace? Is there a reason why this is changed?

Reply:
Yes, but a container name only needs to be unique inside the same pod. And it makes it easier if the name is the same for all releases (f.e. for things like `kubectl logs --container rocketchat`)